src/lib/EventPublisher.js

```diff
@@ -56,13 +57,17 @@ type Subscription = {
 class EventPublisher extends EventEmitter {
   _subscriptionsByID: Map<string, Subscription> = new Map();
 
-  publish = (eventData: EventData) => {
-    const ttl = eventData.ttl && eventData.ttl > 0
-      ? eventData.ttl
-      : settings.DEFAULT_EVENT_TTL;
+  publish = (eventData: EventData, options: PublishOptions) => {
+    const { isInternal, isPublic } = options || {};
```

Comment:
this incorrect: 
1) if `options: PublishOptions`, we should always have them in args and `|| {}` becomes useless
2) do destruction inside function args.